File: pdoo/ruby/betapowerefficientspacestation.rb
#encoding:utf-8

require_relative "dice"
require_relative "powerefficientspacestation"

module Deepspace

  class BetaPowerEfficientSpaceStation<PowerEfficientSpaceStation
    
    @@EXTRAEFFICIENCY = 1.2

    def initialize(station)
      newCopy(station)
    end

    def fire
      dice = Dice.new
      if dice.extraEfficiency
        return super*@@EXTRAEFFICIENCY
      else
        return super
      end
    end
  end
end

File: pdoo/java/Deepspace/BetaPowerEfficientSpaceStation.java
/**
 * @author Manuel Gachs Ballegeer
 */
package Deepspace;

/**
 * @brief Clase nave espacial más eficiente
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    private static final float EXTRAEFFICIENCY = 1.2f;
    
    BetaPowerEfficientSpaceStation(SpaceStation station) {
        super(station);
    }
    
    @Override
    public float fire() {
        Dice dice = new Dice();
        if (dice.extraEfficiency())
            return super.fire()*EXTRAEFFICIENCY;
        else
            return super.fire();
    }
    
    @Override
    public BetaPowerEfficientSpaceStationToUI getUIversion() { return new BetaPowerEfficientSpaceStationToUI(this); }
}
